cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1045 @@
+    if method == "objectlab":
+        scores = _get_subtype_label_quality_scores(
+            labels,
+            predictions,
+            alpha=ALPHA,
+            low_probability_threshold=LOW_PROBABILITY_THRESHOLD,
+            high_probability_threshold=HIGH_PROBABILITY_THRESHOLD,
+            temperature=TEMPERATURE,
+        )
+
+    return scores
+
+
+def _get_min_pred_prob(
+    predictions: List[np.ndarray],
+) -> float:
+    """Returns min pred_prob out of all predictions."""
+    pred_probs = [1.0]  # avoid calling np.min on empty array.
+    for prediction in predictions:
+        for class_prediction in prediction:
+            pred_probs.extend(list(class_prediction[:, -1]))
+
+    min_pred_prob = np.min(pred_probs)
+    return min_pred_prob
+
+
+def _prune_by_threshold(
+    predictions: List[np.ndarray], threshold: float, verbose: bool = True
+) -> List[np.ndarray]:
+    """Removes predicted bounding boxes from predictions who's pred_prob is below the cuttoff threshold."""
+
+    predictions_copy = copy.deepcopy(predictions)
+    num_ann_to_zero = 0
+    total_ann = 0
+    for idx_predictions, prediction in enumerate(predictions_copy):
+        for idx_class, class_prediction in enumerate(prediction):
+            filtered_class_prediction = class_prediction[class_prediction[:, -1] >= threshold]
+            if len(class_prediction) > 0:
+                total_ann += 1
+                if len(filtered_class_prediction) == 0:
+                    num_ann_to_zero += 1
+
+            predictions_copy[idx_predictions][idx_class] = filtered_class_prediction
+
+    p_ann_pruned = total_ann and num_ann_to_zero / total_ann or 0  # avoid division by zero
+    if p_ann_pruned > MAX_ALLOWED_BOX_PRUNE:
+        warnings.warn(
+            f"Pruning with threshold=={threshold} prunes {p_ann_pruned}% labels. Consider lowering the threshold.",
+            UserWarning,
+        )
+    if verbose:
+        print(f"Pruning {num_ann_to_zero} labels out of {total_ann} using threshold=={threshold}.")
+    return predictions_copy
+
+
+def _assert_valid_inputs(
+    labels: List[Dict[str, Any]],
+    predictions,
+    method: Optional[str] = None,
+    threshold: Optional[float] = None,
+):
+    """Asserts proper input format."""
+    if len(labels) != len(predictions):
+        raise ValueError(
+            f"labels and predictions length needs to match. len(labels) == {len(labels)} while len(predictions) == {len(predictions)}."
+        )
+    # Typecheck labels and predictions
+    if not isinstance(labels[0], dict):
+        raise ValueError(
+            f"Labels has to be a list of dicts. Instead it is list of {type(labels[0])}."
+        )
+    # check last column of predictions is probabilities ( < 1.)?
+    if not isinstance(predictions[0], np.ndarray):
+        raise ValueError(
+            f"Prediction has to be a list of np.ndarray. Instead it is list of {type(predictions[0])}."
+        )
+    if not predictions[0][0].shape[1] == 5:
```

Comment:
```suggestion
        )
    p = np.array(p[-1] for p in predictions]
    if (np.min(p)>0.0 or np.max(p) < =1.0):
       raise ValueError(
            f"Probabilities must be between 0.0 and 1.0")
    if not predictions[0][0].shape[1] == 5:
```

Reply:
This would not work. `predictions` is a list of `prediction`. Each `prediction` is a list (or np.ndarray) of np.ndarrays per class. 

Taking `p[-1]` would take all the bounding box predictions of the last class, not the probabilities.

Do we want to do this check for all probabilities (would it be too slow)?

Reply:
Write it can become too slow if there are many predicted boxes with a low threshold.
We can ignore this